examples/boids_qtree/boids.js

```diff
@@ -0,0 +1,203 @@
+// Daniel Shiffman
+// http://natureofcode.com
+// Boid class based on the Nature of Code
+// approach to autonomous agents.
+
+// Ported by Ali Al-Saqban
+
+// Boid class that utilizes qtree data-structure
+
+class Boid {
+
+    constructor() {
+        // Initializing location to be spread out rather than starting from one point
+        // which affects performance.
+        this.vel = createVector(random(-1,1),random(-1,1));
+        this.acc = createVector(0, 0);
+        this.loc = createVector(randomGaussian(width/2,4),randomGaussian(height/2,4));
+
+        // Other attributes and limits
+        this.r = 7
+        this.maxspeed = 3;
+        this.maxforce = 0.05;
+        this.mass = 1;
+
+    }
+
+    run(qtree) {
+        // The NoC approach took in a Boids list prior to this. Switch to qtree data-structure instead
+        this.flock(qtree);
+        this.update();
+        this.render();
+        this.torroidal();
+    }
+
+    update() {
+        //Update location
+        this.vel.add(this.acc);
+
+        //Limit velocity and update location
+        this.vel.limit(this.maxspeed);
+        this.loc.add(this.vel);
+
+        //Reset acceleration
+        this.acc.mult(0)
+    }
+
+    applyForce(force) {
+        // Rearranging F = ma to a = F/m
+        this.acc.add(force)
+    }
+
+    seek(target){
+        let desired = p5.Vector.sub(target, this.loc);
+
+        // Normalize and scale
+        desired.normalize();
+        desired.mult(this.maxspeed)
+
+        // Steer = Desired minus Velocity
+        let steer = desired.sub(this.vel);
+
+        // Limit steer vector magnitude (comment it out for some fun arrangements)
+        //steer.limit(this.maxforce);
+
+        return steer
+
+    }
+
+    flock(qtree) {
+        // The flocking trifecta: separation, alignment, and cohesion.
+        // Try these settings out with no background in draw
+        // | 0.2 | 1 | 1.5 | // | 1.5 | 1 | 1.3 | // | 1 | 1 | 1.3 |
+
```

Comment:
Would these values be better represented by variables, that way they can be more easily tweaked programmatically

Reply:
I'll work on making an input box to these values plus some predefined values to check out to make it easier to play with.

Reply:
@Dezzles I also think I'll have two different initial starts for the particles, one that is clustered closer together, and one that is more spread out. Just as a way of illustrating how quadtree performance is affected.